src/Components/LoginPage/Signup.tsx

```diff
@@ -0,0 +1,58 @@
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import {
+  FormContainer,
+  PasswordElement,
+  TextFieldElement,
+} from "react-hook-form-mui";
+import { Button, Stack, Box } from "@mui/material";
+import { createUser } from "~/Helper Functions/ApiCalls";
+import { Link as RouterLink } from "react-router";
+import signupSchema from "~/types/Forms/SignupSchema";
+import SignupFormInputs from "~/types/Forms/SignupFormInputs";
+
+export default function SignupView() {
```

Comment:
```suggestion
export default function SignupView({setUserData}) {
```

Since you are already making a API call and receiving data back, it would be great to  add it to our userData userState.

Reply:
Sure but that should be in the login right not here ?

Reply:
It should be in both places. If a existing user logs in, the userData needs to be set. If a new user is created, the userData also need to be set. If we don't do that, the new User will have to log in after they register. I think it makes sense to automatically log them in after they submit their registration.